### DISCUSSION TITLE
Duplicate forks

### DISCUSSION THREAD

**Discussion by broderickhyman** [mdziekon/OctoPrint-SpoolManager#53] [General]
@mdziekon As you probably noticed we had @dojohnso take over the "official" Spool Manager in the plugins repository. I had noticed that @dojohnso had taken over the Print History, and saw that you hadn't gotten around to submitting the PR for the change of ownership for several months, so I recommended him for the new owner. Unfortunately, I hadn't noticed the scope of the work you had done, I thought it was just a couple bug fixes. I am sorry about that. We probably should have either used your fork or forked from you.

I am curious what both of your thoughts are on this. @mdziekon do you feel like you should be the owner? Would you prefer to not be the owner and we transfer your changes? Or perhaps you would just like to keep your own fork with your changes as a kind of split? Or some other scenario?
For @dojohnso, would you prefer someone else took over the Spool Manager while you keep the Print History? I don't know how connected the two are and whether it matters if they are under the same owner.

P.S. I don't use discussions much so I am not sure if you will both get a notification for this.

**Comment by mdziekon** [comment]
Hi @broderickhyman, thanks for reaching out and sorry for not answering sooner (holiday period and all that stuff...).

It's been a while since I originally decided to maintain my own fork. Given the history of the plugin's take over  (back when it was originally initiated, not the "second attempt" by @dojohnso), and my current interests in the 3d printing ecosystem, I think I'm not longer interested in maintaining this plugin in any official manner, or at least not as the owner. I might contribute further, as I'm still using the plugin myself (although in limited manner), but I don't want to be the person responsible for its future.

---

First, the ecosystem - since around summer of this year, I've heavily invested in Klipper ecosystem, mainly because of my new Voron, where using Octoprint and its plugins does not make much sense. Even though I still use Octoprint with my other printer, I'm slowly thinking about klipperizing it, which would allow me to unify my experience. Even if that won't come to fruition, there's still the problem of unification of spool management, which unfortunately won't be solved with Octoprint Spoolmanager (even with newly added external DB support), as there's no easy way to integrate it into Klipper (ecosystem). On the other hand, there's Spoolman, which not only integrated with Klipper quite well, it also should allow for integration with Octoprint in the future, mostly because of its architecture design (independence of any ecosystem). So, at least for me, investment into Spoolmanager does not make sense anymore, and I could spend that time on helping with Spoolman's integration.

Second, the "take over" - to be honest, I kind of lost my will to do that right after the original author blocked the first take over for absolutely no reason. They've clearly abandoned both of the plugins, did not respond to any bug reports and even pull requests, and then out of the blue stopped it from being taken over (with a lot of people, not only me, willing to support the cause with maintenance) for it to be abandoned again. Why? Who knows... They've promised to contact me about the future actions, but never reached out in any way. It's not the end of the world (as you and @dojohnso clearly shown, with the successful second take over, which undoubtedly is a great thing), but got me kind of demotivated, not only because of the original author's weird action, but also the overall community and plugins repo maintainers'... apathy, I guess. It's great to see the rejuvenated interest, but I think it's a bit too late for me, I've already lost the spark here.

---

So, whining aside, as for the future suggested actions... Looking at current (2024-01-01) diffs, it seems my fork has a lot more changed files compared to the new and official fork. I think rebasing current official fork on top of my fork would be the best possible action, as it would provide with the least amount of conflicts when joining codebases. Also, it will maintain both my and @dojohnso git history, which might be beneficial for future maintainers. This will also maintain any attribution of my work (although, it would also be nice to be mentioned somewhere in the README :D). I guess @dojohnso should also look into bumping the version a bit, to help people who use my fork (I think they won't see the update notification unless the version number will be bumped above what I currently have).

If my answers or suggestions are unclear, feel free to ask further questions.

---

TL;DR:
- No, I'm no longer interested in taking ownership of this plugin (on the plugins repo).
- I would suggest rebasing the current official fork and applying changes from @dojohnso 's fork on top of my work (I think this approach would simply be easier to perform, due to the scope of my changes).
